File: binv11/sixth_problem.js
const fs = require("fs");

const staffs = JSON.parse(fs.readFileSync("input_data/staff.json", "utf8"));

const highestRatedStaff = staffs.flatMap((staff) => {
  // console.log({ staff });

  if (staff.result.rating <= 4) return [];

  return [
    {
      id: staff.id,
      full_name: staff.full_name,
      email: staff.email,
      rating: staff.result.rating,
    },
  ];
});

console.log(JSON.stringify({ data: highestRatedStaff }, null, 2));

File: binsV1/binV2/sixth_problem.rb
require 'json'

input_data = JSON.parse(File.read('input_data/staff.json'))

staffs = input_data.select { |staff| staff['result']['rating'] > 4 }

highest_rated_staff = staffs.map do |staff|
  {
    id: staff['id'],
    full_name: staff['full_name'],
    email: staff['email'],
    rating: staff['result']['rating']
  }
end

print JSON.pretty_generate({ data: highest_rated_staff })
